app/src/main/java/dev/arkbuilders/arkmemo/utils/SVGUtils.kt

```diff
@@ -0,0 +1,172 @@
+import kotlin.io.path.reader
+import kotlin.io.path.writer
+
+class SVG {
+    var strokeColor = "black"
+    private var fill = "none"
+    private var viewBox = "0${SPACE}0$SPACE$100$SPACE$100"
+    var pathData = ""
+        private set
+
+    private val paint
+        get() = Paint().also {
+            it.color = Color.parseColor(strokeColor)
+            it.style = Paint.Style.STROKE
+            it.strokeWidth = 10f
+            it.strokeCap = Paint.Cap.ROUND
+            it.strokeJoin = Paint.Join.ROUND
+            it.isAntiAlias = true
+        }
+
+    fun writeData(command: String) {
+        pathData += "$SPACE$command"
+    }
+
+    fun moveTo(x: Float, y: Float) = "M$SPACE$x,$y"
+
+    fun relativeLineTo(x: Float, y: Float) = "L$SPACE$x,$y"
+
+    fun quadraticBezierTo(x1: Float, y1: Float, x2: Float, y2: Float) =
+        "Q$SPACE$x1,$y1$SPACE$x2,$y2"
+
+    fun setViewBox(width: Float, height: Float) {
+        viewBox = "0${SPACE}0$SPACE$width$SPACE$height"
+    }
+
+    fun generate(path: Path) {
+        val xmlSerializer = Xml.newSerializer()
+        xmlSerializer.apply {
+            setOutput(path.writer())
+            startDocument("utf-8", false)
+            startTag("", "svg")
+            attribute("", Attributes.VIEW_BOX, viewBox)
+            attribute("", Attributes.XML_NS_URI, XML_NS_URI)
+            startTag("", "path")
+            attribute("", Attributes.Path.STROKE, strokeColor)
+            attribute("", Attributes.Path.FILL, fill)
+            attribute("", Attributes.Path.DATA, pathData)
+            endTag("", "path")
+            endTag("", "svg")
+            endDocument()
+        }
+    }
+
+    fun getPaths() = createPaths()
+
+    fun copy() = SVG().apply {
+        strokeColor = this@SVG.strokeColor
+        fill = this@SVG.fill
+        viewBox = this@SVG.viewBox
+        pathData = this@SVG.pathData
+    }
+
+    private fun createPaths(): Stack<DrawPath> {
+        val paths = Stack<DrawPath>()
+        if (pathData.isNotEmpty()) {
+            var path = AndroidDrawPath()
+            var commandsString = pathData
+            val numberOfCommands = commandsString.filter {
+                it.isLetter()
+            }.length
+            Log.d("svg-utils", "$numberOfCommands commands available")
+            for (index in 1..numberOfCommands) {
+                when (commandsString.substringAfter(SPACE).substringBefore(SPACE)) {
+                    MOVE_TO -> {
+                        path = AndroidDrawPath()
+                        val point = commandsString.substringAfter("$MOVE_TO$SPACE")
+                            .substringBefore(SPACE)
+                        val pointList = point.split(COMMA)
+                        val x = pointList[0].toFloat();
+                        val y = pointList[1].toFloat()
+                        Log.d("svg-utils", "move to $point")
+                        commandsString = commandsString.removePrefix("$SPACE$MOVE_TO$SPACE$point")
```

Comment:
Please pay attention to these lint issues.

![Screenshot 2023-11-10 at 21 25 20](https://github.com/ARK-Builders/ARK-Memo/assets/4281969/15346e6e-dea7-47b5-ba4a-c6fc01c8b678)